File: android_smart_glasses/main/app/src/main/java/com/wearableintelligencesystem/androidsmartglasses/comms/gatt/BleConfig.java
package com.wearableintelligencesystem.androidsmartglasses.comms.gatt;

import static com.wearableintelligencesystem.androidsmartglasses.comms.gatt.Constants.CONFIRM_UUID;
import static com.wearableintelligencesystem.androidsmartglasses.comms.gatt.Constants.SERVER_REC_MESSAGE_UUID;
import static com.wearableintelligencesystem.androidsmartglasses.comms.gatt.Constants.SERVER_SEND_MESSAGE_UUID;
import static com.wearableintelligencesystem.androidsmartglasses.comms.gatt.Constants.SERVICE_UUID;

import java.util.UUID;

public class BleConfig {
    private static final String TAG = "BleConfig";

    public final UUID serviceUUID;
    public final  UUID messageUUID;
    public final  UUID recMessageUUID;
    public final  UUID confirmUUID;

    public BleConfig(UUID serviceUUID, UUID messageUUID, UUID recMessageUUID, UUID confirmUUID) {
        this.serviceUUID = serviceUUID;
        this.messageUUID = messageUUID;
        this.recMessageUUID = recMessageUUID;
        this.confirmUUID = confirmUUID;
    }

    public static class Builder {
        private UUID serviceUUID = SERVICE_UUID;
        private UUID messageUUID = SERVER_SEND_MESSAGE_UUID;
        private UUID recMessageUUID = SERVER_REC_MESSAGE_UUID;
        private UUID confirmUUID = CONFIRM_UUID;

        public Builder setServiceUuid(String uuid) {
            this.serviceUUID = UUID.fromString(uuid);
            return this;
        }

        public Builder setMessageUUid(String uuid) {
            this.messageUUID = UUID.fromString(uuid);
            return this;
        }

        public Builder setConfirmUuid(String uuid) {
            this.confirmUUID = UUID.fromString(uuid);
            return this;
        }

        public Builder setReceiveMessageUUID(String uuid) {
            this.recMessageUUID = UUID.fromString(uuid);
            return this;
        }

        public BleConfig build() {
            return new BleConfig(serviceUUID, messageUUID, recMessageUUID, confirmUUID);
        }
    }

}

File: android_smart_phone/main/app/src/main/java/com/smartglassesmanager/androidsmartphone/smartglassescommunicators/gatt/BleConfig.kt
package com.smartglassesmanager.androidsmartphone.smartglassescommunicators.gatt


import java.util.UUID

class BleConfig(
    val serviceUUID: UUID,
    val serverSendMessageUUID: UUID,
    var serverReceiveMessageUUID: UUID,
    val confirmUUID: UUID,
) {

    class Builder {
        var serviceUUID: UUID = SERVICE_UUID
        var serverSendmessageUUID: UUID = SERVER_SEND_MESSAGE_UUID
        var serverReceiveMessageUUID :UUID = SERVER_REC_MESSAGE_UUID
        var confirmUUID: UUID = CONFIRM_UUID
        fun setServiceUuid(uuid: String): Builder {
            serviceUUID = UUID.fromString(uuid)
            return this
        }

        fun setServerSendMessageUUid(uuid: String): Builder {
            serverSendmessageUUID = UUID.fromString(uuid)
            return this
        }

        fun setConfirmUuid(uuid: String): Builder {
            confirmUUID = UUID.fromString(uuid)
            return this
        }

        fun setServerReceiveMessageUUID(uuid:String ):Builder{
            serverReceiveMessageUUID = UUID.fromString( uuid )
            return this
        }

        fun build(): BleConfig {
            return BleConfig(
                serviceUUID,
                serverSendmessageUUID,
                serverReceiveMessageUUID,
                confirmUUID
            )
        }

    }

    companion object {
        private const val TAG = "BleConfig"
    }
}